doc/LTI_TOOL.md

```diff
@@ -0,0 +1,109 @@
+A+ LTI 1.3 tool configuration
+=============================
+
+## Generating RSA keys
+
+Using A+ as an LTI 1.3 tool requires generating an RSA keypair, with the private key being in pkcs8 format. A valid keypair can be generated with openssl using the following commands:
+
+```
+openssl genrsa -out orig.pem 2048
+openssl rsa -in orig.pem -pubout -out public.key
+openssl pkcs8 -topk8 -inform PEM -outform PEM -nocrypt -in orig.pem -out private.key
+rm orig.pem
+```
+
+
+## Settings in local development
+
+For local development, settings in e.g. `local_settings.py` can be used. Copy-pasteable example in `lti_local_settings.py`. Still need to change the three `client_id` entries after adding the tool settings in moodle.
+
+The following changes should be made in aplus-manual's docker-compose file.
+
+Under `volumes`, add:
+```
+  moodledata:
+  moodledb:
+```
+
+Under `services`, add moodle:
+```
+  moodle:
+    image: apluslms/run-moodle-astra:1.11-3.11
+    depends_on:
+      - db
+      - grader
+    environment:
+      MOODLE_DOCKER_DBTYPE: pgsql
+      MOODLE_DOCKER_DBNAME: moodle
+      MOODLE_DOCKER_DBUSER: moodle
+      MOODLE_DOCKER_DBPASS: "m@0dl3ing"
+      MOODLE_DOCKER_BROWSER: firefox
+      MOODLE_DOCKER_WEB_HOST: "moodle"
+      MOODLE_DOCKER_WEB_PORT: "8050"
+    ports:
+      - "8050:8050"
+    volumes:
+      - moodledata:/var/www/moodledata
+      - ./apache-config/000-default.conf:/etc/apache2/sites-enabled/000-default.conf
+      - ./apache-config/ports.conf:/etc/apache2/ports.conf
+  db:
+    image: postgres:12
+    environment:
+      POSTGRES_USER: moodle
+      POSTGRES_PASSWORD: "m@0dl3ing"
+      POSTGRES_DB: moodle
+    volumes:
+      - moodledb:/var/lib/postgresql/data
+```
+And the following line in `services.plus.environment`:
+```
+APLUS_LOCAL_SETTINGS: '/srv/aplus/a-plus/local_settings.py'
```

Comment:
Is this file path correct? There is no such file in the A-plus git repo. Does it mean that one should create this file oneself based on the new file `doc/lti_local_settings.py`?

Reply:
Yes, this is the intention - e.g. when mounting the user's own code, they should have the local_settings.py file there as well.